code/ship/ship.cpp

```diff
@@ -12458,9 +12458,9 @@ int ship_fire_primary(object * obj, int force, bool rollback_shot)
 
 		// lets start gun convergence / autoaim code from here - Wanderer
 		// in order to do this per weapon, this needs to be moved here from above -Mjn
-		has_converging_autoaim = ((sip->aiming_flags[Ship::Aiming_Flags::Autoaim_convergence] || (The_mission.ai_profile->player_autoaim_fov[Game_skill_level] > 0.0f && !( Game_mode & GM_MULTIPLAYER ))) && aip->target_objnum != -1);
-		has_autoaim = ((has_converging_autoaim || (sip->aiming_flags[Ship::Aiming_Flags::Autoaim]) || (winfo_p->autoaim_fov > 0.0f) || (sip->bank_autoaim_fov[bank_to_fire] > 0.0f)) && aip->target_objnum != -1);
-		needs_target_pos = ((has_autoaim || (sip->aiming_flags[Ship::Aiming_Flags::Auto_convergence])) && aip->target_objnum != -1);
+		has_converging_autoaim = ((sip->aiming_flags[Object::Aiming_Flags::Autoaim_convergence] || winfo_p->aiming_flags[Object::Aiming_Flags::Autoaim_convergence] || (The_mission.ai_profile->player_autoaim_fov[Game_skill_level] > 0.0f && !( Game_mode & GM_MULTIPLAYER ))) && aip->target_objnum != -1);
+		has_autoaim = ((has_converging_autoaim || (sip->aiming_flags[Object::Aiming_Flags::Autoaim]) || winfo_p->aiming_flags[Object::Aiming_Flags::Autoaim] || (sip->bank_autoaim_fov[bank_to_fire] > 0.0f)) && aip->target_objnum != -1);
+		needs_target_pos = ((has_autoaim || has_converging_autoaim) && aip->target_objnum != -1); // This change needs some scrutiny but it seems right?
```

Comment:
Good question. In current master this part of the check is `(sip->aiming_flags[Ship::Aiming_Flags::Auto_convergence])` and in the code here it is replaced with `has_converging_autoaim`.

`(sip->aiming_flags[Ship::Aiming_Flags::Auto_convergence])` is convergence with no autoaim on, and thus not the same as `has_converging_autoaim`. I would say using `(sip->aiming_flags[Object::Aiming_Flags::Auto_convergence]) || winfo_p->aiming_flags[Object::Aiming_Flags::Auto_convergence])` would fit as a more proper replacement. (Unless I'm missing something). What do you think?

Reply:
Ah yes. Good catch. I conflated auto_convergence with autoaim_convergence!

Reply:
Happy to help!

Reply:
I'm going to break down these bools for readability, I think. There's just a lot of checks here. The complier should optimize these out anyway.

Reply:
Ok, give that a look. I think that's a lot more readable

Reply:
Thanks, that's definitely a good change.